test_conformance/extensions/cl_khr_command_buffer/negative_command_buffer_copy.cpp

```diff
@@ -0,0 +1,653 @@
+        test_error(error, "BasicCommandBufferTest::SetUp failed");
+
+        context1 = clCreateContext(0, 1, &device, nullptr, nullptr, &error);
+        test_error(error, "Failed to create context");
+
+        image_ctx = create_image_2d(context1, CL_MEM_READ_WRITE, &formats, 512,
+                                    512, 0, NULL, &error);
+        test_error(error, "create_image_2d failed");
+
+        buffer_ctx = clCreateBuffer(context1, CL_MEM_READ_WRITE, data_size(),
+                                    nullptr, &error);
+        test_error(error, "Unable to create buffer");
+
+
+        return CL_SUCCESS;
+    }
+
+    cl_int Run() override
+    {
+        cl_int error = clCommandCopyImageToBufferKHR(
+            command_buffer, nullptr, image_ctx, buffer, origin, region, 0, 0,
+            nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_CONTEXT,
+                               "clCommandCopyImageToBufferKHR should return "
+                               "CL_INVALID_CONTEXT",
+                               TEST_FAIL);
+
+
+        error = clCommandCopyImageToBufferKHR(command_buffer, nullptr, image,
+                                              buffer_ctx, origin, region, 0, 0,
+                                              nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_CONTEXT,
+                               "clCommandCopyImageToBufferKHR should return "
+                               "CL_INVALID_CONTEXT",
+                               TEST_FAIL);
+
+
+        return CL_SUCCESS;
+    }
+    clMemWrapper image_ctx = nullptr;
+    clMemWrapper buffer_ctx = nullptr;
+    clContextWrapper context1 = nullptr;
+};
+
+// CL_INVALID_SYNC_POINT_WAIT_LIST_KHR if sync_point_wait_list is NULL and
+// num_sync_points_in_wait_list is > 0,
+// or sync_point_wait_list is not NULL and num_sync_points_in_wait_list is 0,
+// or if synchronization-point objects in sync_point_wait_list are not valid
+// synchronization-points.
+struct CommandBufferCopyBufferSyncPointsNullOrNumZero
+    : public CommandBufferCopyBaseTest<false>
+{
+    using CommandBufferCopyBaseTest::CommandBufferCopyBaseTest;
+
+    cl_int Run() override
+    {
+        cl_sync_point_khr invalid_point = 0;
+
+        cl_int error = clCommandCopyBufferKHR(command_buffer, nullptr, in_mem,
+                                              out_mem, 0, 0, data_size(), 1,
+                                              &invalid_point, nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_SYNC_POINT_WAIT_LIST_KHR,
+                               "clCommandCopyBufferKHR should return "
+                               "CL_INVALID_SYNC_POINT_WAIT_LIST_KHR",
+                               TEST_FAIL);
+
+        error = clCommandCopyBufferRectKHR(
+            command_buffer, nullptr, in_mem, out_mem, origin, origin, region, 0,
+            0, 0, 0, 1, &invalid_point, nullptr, nullptr);
```

Comment:
I'm seeing `CL_INVALID_VALUE` here with OCK based rather than ` CL_INVALID_SYNC_POINT_WAIT_LIST_KHR` , based on [the check](https://github.com/codeplaysoftware/oneapi-construction-kit/blob/main/source/cl/source/validate.cpp#L75) that 
```
  OCL_CHECK((((src_origin[2] + region[2] - 1) * src_slice_pitch) +
             ((src_origin[1] + region[1] - 1) * src_row_pitch) +
             (src_origin[0] + region[0])) > src_buffer->size,
            return CL_INVALID_VALUE);
```
where 
```
(gdb) print ((src_origin[2] + region[2] - 1) * src_slice_pitch)                                                        
 $11 = 0
(gdb) print ((src_origin[1] + region[1] - 1) * src_row_pitch)
$12 = 261632
(gdb) print src_origin[0] + region[0]
$14 = 512
(gdb) print src_buffer->size
$13 = 65536
```

I need to familiarize myself with [clEnqueueCopyBufferRect](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#clEnqueueCopyBufferRect) but it may be that we're using the buffer copy in an invalid way.

Reply:
https://github.com/KhronosGroup/OpenCL-CTS/pull/1969 has an image fix that could be used for inspiration